test/unit/nox/_format_test.py

```diff
@@ -21,11 +102,71 @@ def test_mode_fix(nox_session, file_with_unneeded_import):
         assert file_with_unneeded_import.read_text() == ""
 
     @staticmethod
-    def test_mode_check(nox_session, file_with_unneeded_import):
+    def test_mode_check(nox_session, file_with_unneeded_import, caplog):
         with pytest.raises(CommandFailed):
             _ruff(
                 session=nox_session,
                 mode=Mode.Check,
                 files=[str(file_with_unneeded_import)],
             )
+        assert (
+            caplog.messages[1]
+            == f"Command ruff check {file_with_unneeded_import} failed with exit code 1"
+        )
         assert file_with_unneeded_import.read_text() == "import black"
+
+
+@pytest.fixture
+def file_with_multiple_problems(tmp_path):
+    file_path = tmp_path / "dummy_file.py"
+    text = """
+    import numpy as np
+    import isort # unused import
+    from typing import Union # outdated typing from < Python 3.10
+    x: Union[int, str]=2 # outdated typing from < Python 3.10
+    y: np.ndarray = np.array([1, 2, 3])
+    """
+    file_path.write_text(cleandoc(text))
+    return file_path
+
+
+def test_project_fix(nox_session, tmp_path, file_with_multiple_problems):
+    with patch("exasol.toolbox.nox._format.PROJECT_CONFIG") as config:
```

Comment:
I think we should add a more descriptive docstring, here.
I personally was wondering:
* What are the problems in the simulated input file?
* Which problems are expected to be fixed, in which way?

Additionally, the comment "# skip version check as modifies version.py file only" made me wonder. 

Maybe "# Simulate version is up-to-date, as version check is out of the scope of the test case." is better?